src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java

```diff
@@ -132,9 +130,27 @@ public Orientation getOrientation() {
     }
 
     private void readOrientationFromEXIF() {
+        Field field = exif.getField(Tag.ORIENTATION);
         Object value = exif.getValue(Tag.ORIENTATION);
-        if (value != null) {
-            orientation = Orientation.forEXIFOrientation((int) value);
+        if (field != null && value != null) {
+            switch (field.getDataType()) {
+                case LONG:
+                case SLONG:
+                case SHORT:
+                case SSHORT:
+                    // According to spec the orientation must be an unsigned short (16 bit)
+                    // However, we have seen exif data in the wild with LONG and SLONG types
+                    // Thus to be lenient we accept either and convert to int (github issue #548)
+                    if (value instanceof Long) {
+                        orientation = Orientation.forEXIFOrientation(Math.toIntExact((long) value));
+                    } else if (value instanceof Integer) {
```

Comment:
hi @janhoy question about the `else if`. How would be the `value instanceof Integer` if the field is already instance of Long? Is that a Java typing issue when the Value is read from EXIF? This is just me not knowing enough probably. Thanks!

Reply:
I was also surprised by this. Looks like some bug somewhere, that the EXIF typing is long but it gets stuffed into an integer, thus causing ClassCastException. That's the kind of stuff you see in the wild even if it is not as per spec.

So we could really skip the reading of EXIF type and just check the Java type of the value, since we need to fall back to that anyway. The only value in the four switch-case clauses is to trigger the warn log msg, but it could also have been in an `else` clause after `instanceof` checks of Long and Integer fails.

Reply:
@janhoy it would be helpful to put those assumptions in a comment

Reply:
@janhoy We are quite happy with this contribution.  We'd like to get this merged for an upcoming release. Are you able to add these code comments?

Reply:
I added three lines of java comment already about this, what execlty do you feel is unclear and could you propose additional comment text to fix it?

> // According to spec the orientation must be an unsigned short (16 bit)
// However, we have seen exif data in the wild with LONG and SLONG types
// Thus to be lenient we accept either and convert to int (github issue #548)

Reply:
Added these two lines:
> // It could also happen that 'value' is in fact a Java Integer even if the
// exif data type is LONG or SLONG, so we need to check for that as well.

Hope this is mergeable and will be included in next release!